test/Bounty.test.js

```diff
@@ -1,113 +1,89 @@
-    function wrappedHandler (...args) {
-      Promise.resolve(handler(...args)).then(resolve).catch(reject);
-    }
+const reward = web3.toWei(1, 'ether');
 
-    event.watch(wrappedHandler);
-  });
-}
+contract('Bounty', function ([_, owner, researcher]) {
+  describe('Against secure contract', function () {
+    beforeEach(async function () {
+      this.bounty = await SecureTargetBounty.new({ from: owner });
+    });
 
-contract('Bounty', function (accounts) {
-  it('sets reward', async function () {
-    let owner = accounts[0];
-    let reward = web3.toWei(1, 'ether');
-    let bounty = await SecureTargetBounty.new();
-    await sendReward(owner, bounty.address, reward);
+    it('sets reward', async function () {
+      await sendReward(owner, this.bounty.address, reward);
 
-    const balance = await ethGetBalance(bounty.address);
-    assert.equal(reward, balance.toNumber());
-  });
+      const balance = await ethGetBalance(this.bounty.address);
+      balance.should.be.bignumber.eq(reward);
+    });
 
-  it('empties itself when destroyed', async function () {
-    let owner = accounts[0];
-    let reward = web3.toWei(1, 'ether');
-    let bounty = await SecureTargetBounty.new();
-    await sendReward(owner, bounty.address, reward);
+    it('empties itself when destroyed', async function () {
+      await sendReward(owner, this.bounty.address, reward);
 
-    const balance = await ethGetBalance(bounty.address);
-    assert.equal(reward, balance.toNumber());
+      const balance = await ethGetBalance(this.bounty.address);
+      balance.should.be.bignumber.eq(reward);
 
-    await bounty.destroy();
-    const updatedBalance = await ethGetBalance(bounty.address);
-    assert.equal(0, updatedBalance.toNumber());
-  });
+      await this.bounty.destroy({ from: owner });
+
+      const updatedBalance = await ethGetBalance(this.bounty.address);
+      updatedBalance.should.be.bignumber.eq(0);
+    });
 
-  describe('Against secure contract', function () {
     it('cannot claim reward', async function () {
-      let owner = accounts[0];
-      let researcher = accounts[1];
-      let reward = web3.toWei(1, 'ether');
-      let bounty = await SecureTargetBounty.new();
-      let event = bounty.TargetCreated({});
-
-      let watcher = async function (err, result) {
-        event.stopWatching();
-        if (err) { throw err; }
-
-        var targetAddress = result.args.createdAddress;
-        await sendReward(owner, bounty.address, reward);
-
-        const balance = await ethGetBalance(bounty.address);
-        assert.equal(reward, balance.toNumber());
-
-        try {
-          await bounty.claim(targetAddress, { from: researcher });
-          assert.isTrue(false); // should never reach here
-        } catch (error) {
-          let reClaimedBounty = await bounty.claimed.call();
-          assert.isFalse(reClaimedBounty);
-        }
-        try {
-          await bounty.withdrawPayments({ from: researcher });
```

Comment:
no idea what this was testing; the transaction actually goes through but obviously withdraws 0 wei. this is covered in pullpayment test, so I removed it here